### PR TITLE
feat(LogoRainbow): Custom Marriage Equality logo

### PR DESCRIPTION
**Pull request by tamoore**
This adds a new RainbowLogo component to the styleguide. 

A couple of things, the actual dimensions of the bars are still being determined, the colors are however correct. Have also dropped the rainbow colors on mobile just because of the issue with regards to the blue background (waiting on design feedback).

If looking at it on surge, I have added the RainbowLogo as the custom logo component on the header. It should allow you to click the checkbox to display the rainbow logo in situ.

RFC URL: N/A

## Commit Message For Review

REASON FOR CHANGE:

To add a logo variant that shows SEEK's support for marriage equality in Australia.
 
EXAMPLE USAGE:

```js
import { Header, LogoRainbow } from 'seek-style-guide/react';

<Header {...} logoComponent={LogoRainbow} />
```

### CODE REVIEW

**Comment by tamoore** [comment]
Actually, as a FYI, may be a bit of tweaking to the colours and bars in this PR, but then general structure won't change much.

File: react/index.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ export { default as Footer } from './Footer/Footer';
 export { default as Header } from './Header/Header';
 export { default as Loader } from './Loader/Loader';
 export { default as Logo } from './Logo/Logo';
+export { default as RainbowLogo } from './RainbowLogo/RainbowLogo';
```
Comment (markdalgleish) [inline react/index.js]: Can we call this `Logo_Rainbow` to make the alphabetical ordering a bit cleaner?

File: react/RainbowLogo/RainbowLogo.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    [styles.root]: true,
+    [styles.invert]: invert
+  });
+  const type = compact && 'compact' || 'default';
+  const height = getViewPortDimension(type, 'height');
+  const width = getViewPortDimension(type, 'width');
+
+  return (
+    <div {...restProps}>
+
```
Comment (markdalgleish) [inline react/RainbowLogo/RainbowLogo.js]: Nitpick, but this is an extra line-break compared to the original `Logo` component, which makes comparison a little harder.

**Comment by markdalgleish** [comment]
@tamoore just a heads up that I've tweaked the PR title and description, changed the usage example to show what consumers need to do in their apps.

**Comment by markdalgleish** [comment]
Code looks good. Obviously worth getting a second opinion on the design across both desktop and mobile before merging.

**Comment by tamoore** [comment]
Ok, so I changed it a lil to include the white dots on mobile. Sorry about this, but a late realisation that I could use the svg that Tim and Cam had provided came to mind.
@markdalgleish let me know if you're ok with it.

**Comment by markdalgleish** [comment]
@tamoore sounds okay to me, if it's easier.

**Pull request merged**